framework/Optimizers/OptimizerBase.py

```diff
@@ -0,0 +1,515 @@
+      Whatever is a temporary value in the class and not inherited from the parent class should be mentioned here
+      The information is passed back in the dictionary
+      @ In, None
+      @ Out, paramDict, dict, dictionary containing the parameter names as keys
+                              and each parameter's initial value as the dictionary values
+    """
+    paramDict = Sampler.getCurrentSetting(self)
+    paramDict.pop('counter', None)
+    paramDict['counter_mdlEval'       ] = self.counter['mdlEval']
+    paramDict['counter_varsUpdate'    ] = self.counter['varsUpdate']
+    paramDict.update(self.localGetCurrentSetting())
+    return paramDict
+
+  def initialize(self,externalSeeding=None,solutionExport=None):
+    """
+      This function should be called every time a clean optimizer is needed. Called before takeAstep in <Step>
+      @ In, externalSeeding, int, optional, external seed
+      @ In, solutionExport, DataObject, optional, a PointSet to hold the solution
+      @ Out, None
+    """
+    # NOTE: counter['varsUpdate'] needs to be set AFTER self.optTraj length is set by the sampler (if used exclusively)
+    self.counter['mdlEval'] = 0
+    self.counter['varsUpdate'] = [0]*len(self.optTraj)
+    self.optTrajLive = copy.deepcopy(self.optTraj)
+    # TODO: We should use retrieveObjectFromAssemblerDict to get the Instance
+    self.mdlEvalHist = self.assemblerDict['TargetEvaluation'][0][3]
+    # check if the TargetEvaluation feature and target spaces are consistent
+    ins  = self.mdlEvalHist.getVars("input")
+    outs = self.mdlEvalHist.getVars("output")
+    for varName in self.fullOptVars:
+      if varName not in ins:
+        self.raiseAnError(RuntimeError,"the optimization variable "+varName+" is not contained in the TargetEvaluation object "+self.mdlEvalHist.name)
+    if self.objVar not in outs:
+      self.raiseAnError(RuntimeError,"the optimization objective variable "+self.objVar+" is not contained in the TargetEvaluation object "+self.mdlEvalHist.name)
+    self.solutionExport = solutionExport
+    if self.solutionExport is None:
+      self.raiseAnError(IOError,'The results of optimization cannot be obtained without a SolutionExport defined in the Step!')
+
+    if type(solutionExport).__name__ not in ["PointSet","DataSet"]:
+      self.raiseAnError(IOError,'solutionExport type must be a PointSet or DataSet. Got '+\
+                                 type(solutionExport).__name__+ '!')
+    # TODO: We should use retrieveObjectFromAssemblerDict to get the Instance
+    if 'Function' in self.assemblerDict.keys():
+      self.constraintFunction = self.assemblerDict['Function'][0][3]
+      if 'constrain' not in self.constraintFunction.availableMethods():
+        self.raiseAnError(IOError,'the function provided to define the constraints must have an implemented method called "constrain"')
+
+    # TODO a bunch of the gradient-level trajectory initializations should be moved here.
```

Comment:
Were the generic initializations from the other optimizers moved here? Does this TODO still apply?